src/oc_erchef/apps/oc_chef_authz/src/oc_chef_authz_acl.erl

```diff
@@ -270,18 +275,30 @@ safe_fetch_ids(_Type, _OrgId, []) ->
     % Save the cost of a query for an empty list -
     % fairly common for users and to a lesser extent groups.
     [];
-safe_fetch_ids(Type, OrgId, Names) ->
-    Records = oc_chef_authz_db:authz_records_by_name(Type, OrgId, Names),
-    FoundNames = lists:sort(names_from_records(Records)),
-    GivenNames = lists:sort(Names),
-    case GivenNames -- FoundNames of
+safe_fetch_ids(Type, OrgId, Names) when not is_list(OrgId) ->
+    safe_fetch_ids(Type, [OrgId], Names);
+safe_fetch_ids(Type, OrgIds, Names) ->
+    {Missing, Records} = safe_fetch_ids_rec(Type, OrgIds, Names, []),
+    case Missing of
         [] ->
             ok;
-        Missing ->
+        _ ->
             throw({invalid, Type, Missing})
     end,
     ids_from_records(Records).
 
+% Recursively expands the types for each org id, removing the found names 
+safe_fetch_ids_rec(_Type, [], Names, Records) ->
+    {Names, Records};
+safe_fetch_ids_rec(_Type, _OrgIds, [], Records) ->
+    {[], Records};
+safe_fetch_ids_rec(Type, [OrgId | OrgIds], Names, RecordsAcc) ->
+    Records = oc_chef_authz_db:authz_records_by_name(Type, OrgId, Names),
+    FoundNames = lists:sort(names_from_records(Records)),
+    GivenNames = lists:sort(Names),
```

Comment:
Could we keep this sort in `safe_fetch_ids` so that we don`t end up doing it more than once? (assuming that `GivenNames--FoundNames` below will preserve the sort order)